boards/Makefile.common

```diff
@@ -124,34 +127,35 @@ target:
 doc: | target
 	$(Q)RUSTDOCFLAGS=--document-private-items $(CARGO) doc $(VERBOSE) --release --target=$(TARGET)
 
-target/$(TARGET)/release/$(PLATFORM).elf: target/$(TARGET)/release/$(PLATFORM)
-	$(Q)cp target/$(TARGET)/release/$(PLATFORM) target/$(TARGET)/release/$(PLATFORM).elf
+target/$(TARGET)/release/%.elf: target/$(TARGET)/release/%
+	$(Q)cp $< $@
 
 target/$(TARGET)/release/$(PLATFORM).lst: target/$(TARGET)/release/$(PLATFORM).elf
-	$(Q)$(OBJDUMP) $(OBJDUMP_FLAGS) $< > target/$(TARGET)/release/$(PLATFORM).lst
+	$(Q)$(OBJDUMP) $(OBJDUMP_FLAGS) $< > $@
 
 .PHONY: target/$(TARGET)/release/$(PLATFORM)
 target/$(TARGET)/release/$(PLATFORM):
 	$(Q)RUSTFLAGS="$(RUSTFLAGS_FOR_CARGO_LINKING)" $(CARGO) build --target=$(TARGET) $(VERBOSE) --release
 	$(Q)$(SIZE) $@
 
 target/$(TARGET)/debug/$(PLATFORM).elf: target/$(TARGET)/debug/$(PLATFORM)
-	$(Q)cp target/$(TARGET)/debug/$(PLATFORM) target/$(TARGET)/debug/$(PLATFORM).elf
+	$(Q)cp $< $@
 
-target/$(TARGET)/debug/$(PLATFORM).lst: target/$(TARGET)/debug/$(PLATFORM).elf
-	$(Q)$(OBJDUMP) $(OBJDUMP_FLAGS) $< > target/$(TARGET)/debug/$(PLATFORM).lst
+target/$(TARGET)/debug/%.lst: target/$(TARGET)/debug/%.elf
+	$(Q)$(OBJDUMP) $(OBJDUMP_FLAGS) $< > $@
 
 .PHONY: target/$(TARGET)/debug/$(PLATFORM)
 target/$(TARGET)/debug/$(PLATFORM):
 	$(Q)RUSTFLAGS="$(RUSTFLAGS_FOR_CARGO_LINKING)" $(CARGO) build $(VERBOSE) --target=$(TARGET)
 	$(Q)$(SIZE) $@
 
-target/$(TARGET)/release/$(PLATFORM).bin: target/$(TARGET)/release/$(PLATFORM).elf
+target/$(TARGET)/release/%.bin: target/$(TARGET)/release/%.elf
 	$(Q)$(OBJCOPY) -O=binary $^ $@
 
-target/$(TARGET)/debug/$(PLATFORM).bin: target/$(TARGET)/debug/$(PLATFORM).elf
+target/$(TARGET)/debug/%.bin: target/$(TARGET)/debug/%.elf
```

Comment:
If we're going to go for pattern rules, why not just go all the way and replace both of these with
`%.bin: %.elf`

Reply:
Right, you're the makefile expert :) I'm entirely sure what might possibly break if I change them that drastically, but if you think it's ok, I'm game.